packages/aleph-holders/src/domain/main.ts

```diff
@@ -1,14 +1,67 @@
-import { BlockchainChain } from '@aleph-indexer/framework'
-import { IndexerMainDomain } from '@aleph-indexer/framework'
+import { 
+  IndexerMainDomain, 
+  BlockchainChain, 
+  IndexerMainDomainWithDiscovery, 
+  AccountIndexerConfigWithMeta,
+  IndexerMainDomainContext,
+} from '@aleph-indexer/framework'
 import {
   Balance,
   BalanceQueryArgs,
-  ERC20TransferEvent,
-  ERC20TransferEventQueryArgs,
+  Snapshot,
+  SolanaBalance,
+  TransferEvent,
+  TransferEventQueryArgs,
 } from '../types.js'
 import { blockchainTokenContract } from '../utils/index.js'
+import { solanaSnapshot } from './solanaSnapshot.js'
+import { createSnapshotDAL } from '../dal/snapshot.js'
+
+export default class MainDomain 
+  extends IndexerMainDomain 
+  implements IndexerMainDomainWithDiscovery 
+{
+  protected firstSnapshotTimestamp?: number
+
+  constructor(
+    protected context: IndexerMainDomainContext,
+    protected tokenSnapshotDAL = createSnapshotDAL(context.dataPath),
+    // note: Snapshot interval is set to 1 hour
+    protected discoveryInterval = 1000 * 60 * 60,
```

Comment:
This feels way too long in my opinion. Is there no way to instantly recognize newly created ATAs? When users add funds to their wallet and want to stake, they expect our indexers to instantly recognize the added tokens.